cookbook/case_studies/feature_engineering/sqlite_datacleaning/datacleaning_workflow.py

```diff
@@ -0,0 +1,109 @@
+"""
+Data Cleaning Imperative Workflow
+---------------------------------------
+
+This workflow makes use of the feature engineering tasks defined in the other file. 
+We'll build an SQLite3 data cleaning pipeline utilizing these tasks.
+"""
+
+# %%
+# Let's import the libraries.
+import pandas as pd
+from flytekit import Workflow, kwtypes, reference_task
+from flytekit.extras.sqlite3.task import SQLite3Config, SQLite3Task
+from flytekit.types.schema import FlyteSchema
+
+# %%
+# Next, we define the reference tasks. A :py:func:`flytekit.reference_task` references the Flyte tasks that have already been defined, serialized, and registered.
+# The primary advantage of using a reference task is to reduce the redundancy; we needn't define the task(s) again if we have multiple datasets that need to be feature-engineered.
+@reference_task(
+    project="flytesnacks",
+    domain="development",
+    name="sqlite_datacleaning.tasks.mean_median_imputer",
+    version="fast4f51f7895819256f2540a08c97a51194",
+)
+def mean_median_imputer(
+    dataframe: pd.DataFrame,
+    imputation_method: str,
+) -> pd.DataFrame:
+    ...
+
+
+@reference_task(
+    project="flytesnacks",
+    domain="development",
+    name="sqlite_datacleaning.tasks.univariate_selection",
+    version="fast4f51f7895819256f2540a08c97a51194",
+)
+def univariate_selection(
+    dataframe: pd.DataFrame,
+    split_mask: int,
+    num_features: int,
+) -> pd.DataFrame:
+    ...
+
+
+# %%
+# .. note::
+#
+#   The ``version`` varies depending on the version assigned during the task registration process.
+
+# %%
+# Finally, we define an imperative workflow that accepts the two reference tasks we've prototyped above. The data flow can be interpreted as follows:
```

Comment:
Why?

Reply:
What i mean is, we want to generate this workflow dynamically. So for example we can write a new cli (lets say a simple one) where you provide a sql file and then it creates a workflow - registers it and you can run it from thUI

Reply:
e.g
```python
def create_fe_workflow(sql: string) -> Workflow:
   ...

if __name__ == "__main__":
   file = argv.parse()
   sql = read_file(file)
   w = create_fe_workflow(sql)
   control_plane.register(w)
```

Reply:
the register part we can do as a follow up

Reply:
I'll keep this open. Let me add this functionality once registration is integrated into the control plane.